physics/kepler_orbit.hpp

```diff
@@ -18,16 +18,63 @@ using quantities::Angle;
 using quantities::AngularFrequency;
 using quantities::GravitationalParameter;
 using quantities::Length;
+using quantities::SpecificAngularMomentum;
+using quantities::SpecificEnergy;
+using quantities::Speed;
+using quantities::Time;
 
 template<typename Frame>
 struct KeplerianElements final {
-  double eccentricity{};
+  // I. These elements determine the shape and size of the conic.  Two are
+  // needed, from two different numbered categories below.
+  // 1. eccentricity.
+  std::experimental::optional<double> eccentricity;
+  // The following two elements are NaN for elliptic orbits.
+  std::experimental::optional<Angle> asymptotic_true_anomaly;
+  std::experimental::optional<Angle> turning_angle;
```

Comment:
I wonder if the fields after section 1 could be a oneof {eliptic, hyperbolic}.  It's quite annoying to have all these fields that are specific to one shape or another and you have to remember which.